### PR TITLE
Use currency fraction information

### PR DESCRIPTION
**Pull request by etanol**
This extends pull request #117 to also load the currency fraction information from CLDR supplemental data. Once loaded, it adds an option to babel.numbers.format_currency so the decimal positions deduced from the format pattern can be ignored and replaced by those of the currency, as indicated by such supplemental information.

The API remains backwards compatible.

### CODE REVIEW

**Comment by sils1297** [comment]
@etanol thanks for submitting this PR. Is there a reason why you did two PRs with one holding only half of the commits? Can I close https://github.com/mitsuhiko/babel/pull/117 as a duplicate?

To get this PR merged:

1. Please rebase onto current master :)
2. The tests must pass so we can merge it. Can you look into that if the rebase does not suffice?

**Comment by etanol** [comment]
This pull request actually includes two developments:

1. Supporting the parent locale exceptions (bugfix)
2. Adding an option to use locale currency information when formatting price amounts (development)

In the first case, I think it would be nice to have it fixed in 2.0.  Then the second one can go into master (3.0).  Therefore the question is, how do you _repeat_ commits?  By commiting to maintenance branches and merging into master?  Or by commiting into master and cherrypicking to maintenance?

**Comment by sils1297** [comment]
@etanol this project was unmaintained a short while ago and we just took over. THat said we didn't decide that yet, but I guess we'll cherry pick it, at least that's what I'd do. @erickwilder ?

**Comment by etanol** [comment]
I separated the parent locale part in a new pull request against master (#180).  Cherry picking to 2.0 shouldn't be too hard.  Then I'll be able to modify this PR to only include the currency decimals development.

Sorry for the noise.

**Comment by etanol** [comment]
Rebased and reviewed, this should fix (or at least, enable a way to fix) bug #139

**Comment by etanol** [comment]
Updated to improve test coverage.

File: babel/numbers.py [hunk 1]
```diff
@@ -701,8 +719,8 @@ def _format_int(self, value, min, max, locale):
             gsize = self.grouping[1]
         return value + ret
 
-    def _format_frac(self, value, locale):
-        min, max = self.frac_prec
+    def _format_frac(self, value, locale, force_frac=None):
+        min, max = force_frac or self.frac_prec
```
Comment (erickwilder) [inline babel/numbers.py]: Can you change `min` and `max` var names? We should avoid using reserved words as much as possible.

Comment (sils1297) [inline babel/numbers.py]: @erickwilder that would be an unrelated change and should go into an own commit which is better for maintenance

File: tests/test_numbers.py [hunk 2]
```diff
@@ -252,6 +253,13 @@ def test_format_currency():
     assert (numbers.format_currency(1099.98, 'EUR', locale='nl_NL')
             != numbers.format_currency(-1099.98, 'EUR', locale='nl_NL'))
 
+    assert (numbers.format_currency(1099.98, 'JPY', locale='en_US',
+                                    currency_digits=True)
+            == u'\xa51,100')
+    assert (numbers.format_currency(1099.98, 'COP', u'#,##0.00', locale='es_ES',
```
Comment (erickwilder) [inline tests/test_numbers.py]: There're no tests with `currency_digits=False`. I'm assuming that `current_digits=None` is the default and current tests are doing their jobs to ensure the correct behaviour.

File: babel/numbers.py (hunk 1 above)
Comment (erickwilder) [inline babel/numbers.py]: @sils1297 
Ok, I created a new issue just to not loose track of this:
https://github.com/mitsuhiko/babel/issues/187

File: tests/test_numbers.py (hunk 2 above)
Comment (etanol) [inline tests/test_numbers.py]: `currency_digits=False` is the default (you can see that in the diff).  The idea is to maintain syntactic and semantic backwards compatibility with older user code.

**Comment by sils1297** [comment]
awesome, those commit messages please my eyes :)

**Comment by erickwilder** [comment]
I agree with @sils1297 to use `None` as the default argument and letting the called of the function to decide the intended behaviour.

**Comment by etanol** [comment]
Rebased again and inverted the behaviour (i.e. use currency decimal digits by default).

File: babel/numbers.py (hunk 1 above)
Comment (mitsuhiko) [inline babel/numbers.py]: Shadowing builtins is just fine.  Babel already does that in many places.  Please do not change this now because that will just make the API inconsistent.  There is nothing wrong with shadowing builtins as you can easily rebind them if you need them locally.

**Comment by sils1297** [comment]
@etanol sorry, conflicts again. Anything against a healthy rebase?

**Comment by sils1297** [comment]
Looks good to me, merging.

**Comment by sils1297** [comment]
oh wait...

**Comment by etanol** [comment]
Sorry, I meesed up the `CHANGES` file.  Let me review with more time, since I think there is some stuff, other than mine, that is out of place

**Comment by sils1297** [comment]
@etanol merging this after CI approval.
